packages/server/test/performance/proxy_performance.js

```diff
@@ -0,0 +1,336 @@
+        `--user-data-dir=${fs.mkdtempSync(path.join(os.tmpdir(), 'cy-perf-'))}`,
+      ])
+
+      if (testCase.disableHttp2) {
+        args.push('--disable-http2')
+      }
+
+      if (testCase.cyProxy) {
+        args.push(`--proxy-server=http://localhost:${CY_PROXY_PORT}`)
+      }
+
+      if (testCase.upstreamProxy && !testCase.cyProxy) {
+        args.push(`--proxy-server=http://localhost:${PROXY_PORT}`)
+      } else if (testCase.httpsUpstreamProxy && !testCase.cyProxy) {
+        args.push(`--proxy-server=https://localhost:${HTTPS_PROXY_PORT}`)
+      }
+
+      if (testCase.upstreamProxy && testCase.cyProxy) {
+        process.env.HTTP_PROXY = process.env.HTTPS_PROXY = `http://localhost:${PROXY_PORT}`
+      } else if (testCase.httpsUpstreamProxy && testCase.cyProxy) {
+        process.env.HTTP_PROXY = process.env.HTTPS_PROXY = `https://localhost:${HTTPS_PROXY_PORT}`
+      } else {
+        delete process.env.HTTPS_PROXY
+        delete process.env.HTTP_PROXY
+      }
+
+      if (testCase.cyIntercept) {
+        cyServer._onDomainSet(URL_UNDER_TEST)
+      } else {
+        cyServer._onDomainSet('<root>')
+      }
+
+      let cmd = CHROME_PATH
+
+      debug('Launching Chrome: ', cmd, args.join(' '))
+
+      const proc = cp.spawn(cmd, args, {
+        stdio: 'ignore',
+      })
+
+      const runHar = () => {
+        // wait for Chrome to open, then start capturing
+        return Promise.delay(500).then(() => {
+          debug('Trying to connect to Chrome...')
+
+          const harCapturer = HarCapturer.run([
+            URL_UNDER_TEST,
+          ], {
+            port: cdpPort,
+            // wait til all data is done before finishing
+            // https://github.com/cyrus-and/chrome-har-capturer/issues/59
+            postHook: (_, cdp) => {
+              let timeout
+
+              return new Promise((resolve) => {
+                cdp.on('event', (message) => {
+                  if (message.method === 'Network.dataReceived') {
+                    // reset timer
+                    clearTimeout(timeout)
+                    timeout = setTimeout(resolve, 1000)
+                  }
+                })
+              })
+            },
+          })
+
+          return new Promise((resolve, reject) => {
+            harCapturer.on('fail', (_, err) => {
+              return reject(err)
+            })
+
+            harCapturer.on('har', resolve)
+          })
+          .catch((err) => {
+            // sometimes chrome takes surprisingly long, just reconn
+            debug('Chrome connection failed: ', err)
+
+            return runHar()
+          })
+          .then((har) => {
+            proc.kill(9)
+            debug('Received HAR from Chrome')
+            getResultsFromHar(har, testCase)
+
+            const runTime = Number(testCase['Total'].replace('ms', ''))
+
+            expect(runTime).to.be.lessThan(getExpectedRunTime(testCase))
+
+            return Promise.resolve()
```

Comment:
This shouldn't be necessary to have

Reply:
was trying to debug and accidentally left that in, good catch